src/server/auth/server/api_server.go

```diff
@@ -875,7 +875,10 @@ func (a *apiServer) GetRobotToken(ctx context.Context, req *auth.GetRobotTokenRe
 	a.LogReq(req)
 	defer func(start time.Time) { a.LogResp(req, resp, retErr, time.Since(start)) }(time.Now())
 
-	subject := auth.RobotPrefix + req.Robot
+	subject := req.Robot
+	if !strings.HasPrefix(subject, auth.RobotPrefix) {
+		subject = auth.RobotPrefix + subject
```

Comment:
would it make sense to error/trim if a user sends e.g. `robbot:foo`?

Reply:
I'm hesitant to trim the user input arbitrarily, so I made colons illegal in robot names. You can send `robot:rosie` or `rosie` but not `robbot:rosie`.